src/Compilers/CSharp/Portable/Symbols/Source/CustomModifierUtils.cs

```diff
@@ -35,8 +35,7 @@ internal static class CustomModifierUtils
             // have already been compared.
             MethodSymbol constructedSourceMethod = sourceMethod.ConstructIfGeneric(destinationMethod.TypeArguments);
 
-            customModifiers = CustomModifiersTuple.Create(constructedSourceMethod.ReturnTypeCustomModifiers,
-                                                          destinationMethod.ReturnsByRef ? constructedSourceMethod.RefCustomModifiers : ImmutableArray<CustomModifier>.Empty);
+            customModifiers = CustomModifiersTuple.Create(constructedSourceMethod.ReturnTypeCustomModifiers, constructedSourceMethod.RefCustomModifiers);
```

Comment:
> My understanding is that all such implementations should exactly duplicate their overridden methods (CLR-spec), which the new implementation is doing.

Only when ref modifier is present, which might not be the case in error situations. If destination method is not ref returning, it shouldn't have RefCustomModifiers because they have nothing to modify. It is fine to adjust the condition, but it is incorrect to remove conditional logic altogether.